src/app/sections/sections.component.ts

```diff
@@ -136,55 +143,69 @@ export class SectionsComponent implements OnInit {
     }
     else {
       this.enablePricingButton = true;
+      this.dataToFillInForms = {};
+      this.calculatedValues = DEFAULT_CALCULATED_VALUES;
     }
   }
 
   getPricingById() {
-    if (this.atlasId) {
+    if (this.atlasId && !this.dataUpdated) {
       this.http
         .get(`${environment.apiUrl}/Price/GetLoanInputs/${this.atlasId}`)
         .subscribe((response: any) => {
           this.dataToFillInForms = response;
         });
+      return;
     }
-    if (!this.atlasId && this.typeSelected === 'New Loan') {
```

Comment:
If the data has been modified then call the post API call.. if condition below is not required.. Please let me know if this is correct